### ISSUE TITLE
Improve the scope of translations

### ISSUE THREAD

**Issue by anna-geller** [kestra-io/kestra#4613]
### Feature description

Some fields are missing translation fields
![image](https://github.com/user-attachments/assets/28ffed4c-f592-49af-a615-dd594978d693)

![image](https://github.com/user-attachments/assets/2a33c146-d622-417a-af93-b773d387fe5e)

![image](https://github.com/user-attachments/assets/4645625a-0505-46c2-b5f5-bae3ac78d0e0)

and some translations are not used within the UI - here you can see the translation exists but is not used:

![image](https://github.com/user-attachments/assets/66f7552b-e57c-4b16-987f-d4d3ea8f6062)

**Comment by Rafael-Floriano** [comment]
Hi there, may I work on this task?

**Comment by MilosPaunovic** [comment]
@Rafael-Floriano Yes you may, but as I assigned you to another one minutes ago, let's go one by one, when that is completed, I'm perfectly happy with you taking over this one, also.

**Comment by SanjyotX** [comment]
hey there, i would love to work on this

**Comment by Ben8t** [comment]
as Milos comment, @SanjyotX I assigned you to another one minutes ago, let's go one by one, when that is completed, we are happy with you taking over this one, also.

**Comment by jemarray** [comment]
Hello, I'd like to work on this issue for hacktoberfest

**Comment by MilosPaunovic** [comment]
Sure, go ahead @jemarray! 🚀

**Comment by MilosPaunovic** [comment]
Hi @jemarray, is there any progress on this?

**Comment by joaomateusmunuera** [comment]
Hello @MilosPaunovic, can i work in this issue?

**Comment by MilosPaunovic** [comment]
Sure, go ahead @joaomateusmunuera!

**Comment by MilosPaunovic** [comment]
Hi @joaomateusmunuera,

Since there hasn't been any progress on the issue, I've unassigned you for now. If you decide to work on it and open a PR, let me know, and I'll be happy to reassign it to you.

In the meantime, the issue is open for anyone in the community who would like to take it on! 🚀

**Comment by YAI-cs** [comment]
Hi! May I work on this issue?

**Comment by MilosPaunovic** [comment]
Absolutely, go for it @YAI-cs! 🚀

**Comment by MilosPaunovic** [comment]
Hi @YAI-cs,

As there hasn't been any progress or updates on this, I'll unassign you for now. However, if you do open a PR, it will be reviewed, and you'll be reassigned if no one else handles it in the meantime.

Thanks for all the efforts so far! 🍀

**Comment by Ahirshson02** [comment]
Hello, I'd like to clarify the task. The translations are already obtained, and we need to simply apply them by giving the UI components a translation attribute that will connect with the files in the translations folder in ui/src/translations? Investigating this issue a bit and would just like some clarification. Thank you.

**Comment by MilosPaunovic** [comment]
Hi @Ahirshson02,

So, the idea behind this task is to have the strings which are now being `hard-coded` to be used as translation strings - that would mean to add them to `en.json` file and to replace it's usage in the component files.

Work here could be started with the strings Anna added as examples into the description of this issue.

Thanks for your interest in this! 🚀

**Comment by Ahirshson02** [comment]
@MilosPaunovic Thank you for that clarification, that was helpful. May I work on this issue? If so, how soon do you expect it to be finished? I am a university student and one my class projects is to contribute to a an open source project. So any estimation on how long I have to finish this issue would be helpful as I prioritize my various tasks.

**Comment by MilosPaunovic** [comment]
You're welcome!

And yes, you may start working on it, and we usually give a week to assignees to show some progress before un-assign them if none is shown. It doesn't mean you can't continue working on it, it's just becoming open for others to do so, also.

**Comment by Ahirshson02** [comment]
Just to provide an update. I have Kestra running, and am still working adding all the translations. I hope to make a pull request in the next few days. I have failed to noticed an organization method in the en.json on how the keys and values are organized. Is there a particular place in the file each translation should go? I've been adding them to the end of the translation file, but am open to trying to reorganize them if necessary to make the pull request go smoothly.

**Comment by MilosPaunovic** [comment]
Nope, adding missing translations to the end is quite okay for now, thanks!

**Comment by Ahirshson02** [comment]
I am trying to test out the translations and I see there is a python script to generate the translations, however when following the instructions in the .github\actions\generate-translations I receive an error message when trying to run the script that says I need to set the OPENAI environment variable or pass the api key to the client. I don't see any API key I can set, but I assume that is private and I don't have access to it. Do I need to generate the translations, or just fill out en.json and the ui components? If I need to generate the translations, some guidance would be appreciated

**Comment by MilosPaunovic** [comment]
Hi @Ahirshson02,

Just like you've said it, you just need to add them to `en.json` file, and we have automation on GitHub which will cover all the other languages we support by using the English one as a base.

**Comment by MilosPaunovic** [comment]
Hey @Ahirshson02, could we expect a PR soon?

**Comment by Ahirshson02** [comment]
Hi @MilosPaunovic I opened a pull request and have received some what I assume is automated feedback about the backend tests and translation generation tests failing. Does this indicate I need to redo something on my end?

**Comment by MilosPaunovic** [comment]
I'll have a look and give my updates there. Thanks for opening a PR! 🚀

**Comment by MilosPaunovic** [comment]
Thanks to everyone involved in handling this issue! 🚀
